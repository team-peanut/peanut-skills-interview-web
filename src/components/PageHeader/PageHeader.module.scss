@use "../../styles/typography";
@use "../../styles/mediaQueries";

.wrapper {
  background-color: var(--secondary-background-colour);

  > div {
    display: grid;
    grid-template-columns: 70px 1fr 70px;
    align-items: center;
    padding: var(--spacing-small) var(--spacing-regular);

    @include mediaQueries.desktop {
      display: flex;
      justify-content: space-between;
      align-items: center;
      max-width: var(--breakpoint-desktop);
      margin: 0 auto;
      padding: 0 var(--spacing-large);
    }

    > a.menu {
      @include mediaQueries.desktop {
        display: none;
      }
      display: block;
      width: 40px;
      height: 40px;
      background-image: url(./assets/menuMobile.svg);
      background-size: 40px 40px;
      background-repeat: no-repeat;
      font-size: 1px;
      color: transparent;
    }

    > a.logo {
      display: block;
      width: 32px;
      height: 18px;
      background-image: url(./assets/logoMobile.svg);
      background-size: 32px 18px;
      margin: 0 auto;

      @include mediaQueries.desktop {
        width: 88px;
        height: 28px;
        background-image: url(./assets/logoDesktop.svg);
        background-size: 88px 28px;
        margin: 0;
      }

      background-repeat: no-repeat;
      font-size: 1px;
      color: transparent;
    }

    > ul {
      display: none;

      @include mediaQueries.desktop {
        display: flex;
      }
      list-style-type: none;
      margin: 0;
      gap: var(--spacing-large);

      > li {
        > a {
          @include typography.typography-menu-item;
          color: #000000;
          text-decoration: none;
          display: block;
          padding: var(--spacing-regular) 0;
          white-space: nowrap;
        }
      }
    }

    .joinButton {
      @include mediaQueries.desktop {
        display: none;
      }
    }
  }
}
