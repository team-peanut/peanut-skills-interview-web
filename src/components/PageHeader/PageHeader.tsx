import React from 'react'
import classNames from "classnames";

import styles from './PageHeader.module.scss';
import ActionButton from '../ActionButton/ActionButton';

export interface Props {
    className?: string
}

function PageHeader({className}: Props): JSX.Element {
    
    return (
        <header className={classNames(styles.wrapper,className)}>
            <div>
                <a href="https://www.peanut-app.io/" title="Peanut" className={styles.menu}>
                    Menu
                </a>

                <a href="https://www.peanut-app.io/" title="Peanut" className={styles.logo}>
                    Peanut
                </a>
                <ul>
                    <li><a href="/planning-for-a-baby">Planning for a Baby</a></li>
                    <li><a href="/pregnancy">Pregnancy</a></li>
                    <li><a href="/baby">Baby</a></li>
                    <li><a href="/toddler-and-child">Toddler &amp; Child</a></li>
                    <li><a href="/health-and-wellbeing">Health &amp; Wellbeing</a></li>
                    <li><a href="/menopause">Menopause</a></li>
                </ul>
                <ActionButton className={styles.button}>
                    <>Join</>
                </ActionButton>
            </div>
        </header>
    );
};

export default PageHeader;
