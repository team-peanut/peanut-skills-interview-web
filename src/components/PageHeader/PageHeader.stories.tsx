import React from "react";
import { ComponentStory } from "@storybook/react";

import PageHeader, {Props} from "./PageHeader";

export default {
  title: "Components/PageHeader",
  component: PageHeader,
};

const Template: ComponentStory<typeof PageHeader> = (props: Props) => (
  <PageHeader  />
);


export const Primary = Template.bind({});
Primary.args = {
  
};

