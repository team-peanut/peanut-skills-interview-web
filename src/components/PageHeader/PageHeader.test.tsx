import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";


import PageHeader, {Props} from "./PageHeader";


describe("PageHeader", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  
  afterEach(cleanup);

  it("renders as expected", () => {
    const { asFragment } = render(<PageHeader />);
    expect(asFragment()).toMatchSnapshot();
  });

});
