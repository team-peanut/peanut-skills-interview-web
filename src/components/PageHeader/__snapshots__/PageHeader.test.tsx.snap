// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders as expected 1`] = `
<DocumentFragment>
  <header
    class="wrapper"
  >
    <div>
      <a
        class="menu"
        href="https://www.peanut-app.io/"
        title="Peanut"
      >
        Menu
      </a>
      <a
        class="logo"
        href="https://www.peanut-app.io/"
        title="Peanut"
      >
        Peanut
      </a>
      <ul>
        <li>
          <a
            href="/planning-for-a-baby"
          >
            Planning for a Baby
          </a>
        </li>
        <li>
          <a
            href="/pregnancy"
          >
            Pregnancy
          </a>
        </li>
        <li>
          <a
            href="/baby"
          >
            Baby
          </a>
        </li>
        <li>
          <a
            href="/toddler-and-child"
          >
            Toddler & Child
          </a>
        </li>
        <li>
          <a
            href="/health-and-wellbeing"
          >
            Health & Wellbeing
          </a>
        </li>
        <li>
          <a
            href="/menopause"
          >
            Menopause
          </a>
        </li>
      </ul>
      <button
        class="wrapper button"
        type="button"
      >
        Join
      </button>
    </div>
  </header>
</DocumentFragment>
`;
